geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/data/KafkaDataStoreTest.scala

```diff
@@ -188,6 +192,86 @@ class KafkaDataStoreTest extends KafkaContainerTest with Mockito {
+        }
+        consumer must not(beNull)
+        producer must not(beNull)
+
+        try {
+          val sft = SimpleFeatureTypes.createImmutableType("kafka", "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;geomesa.foo='bar'")
+          val topic = s"${producer.config.catalog}-${sft.getTypeName}".replaceAll("/", "-")
+
+          producer.createSchema(sft)
+
+          consumer.metadata.resetCache()
+          foreach(Seq(producer, consumer)) { ds =>
+            ds.getTypeNames.toSeq mustEqual Seq(sft.getTypeName)
+            val schema = ds.getSchema(sft.getTypeName)
+            schema must not(beNull)
+            schema mustEqual sft
+            schema.getUserData.get("geomesa.foo") mustEqual "bar"
+            schema.getUserData.get(KafkaDataStore.TopicKey) mustEqual topic
+          }
+
+          val props = Collections.singletonMap[String, AnyRef](AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
+          WithClose(AdminClient.create(props)) { admin =>
+            admin.listTopics().names().get.asScala must contain(topic)
+          }
+
+
+          val pprops = Map[String, Object](
+            CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG -> brokers,
+            ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG -> classOf[StringSerializer].getName,
+            ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG -> classOf[StringSerializer].getName
+          ).asJava
+          WithClose(new KafkaProducer[String, String](pprops)) { p =>
+            p.send(new ProducerRecord(topic, null, "dummy1"))
+            p.send(new ProducerRecord(topic, null, "dummy2"))
+            p.flush()
+          }
+
+          consumer.removeSchema(sft.getTypeName)
+          foreach(Seq(consumer, producer)) { ds =>
+            eventually(40, 100.millis)(ds.getTypeNames.toSeq must beEmpty)
+            ds.getSchema(sft.getTypeName) must beNull
+          }
+          WithClose(AdminClient.create(props)) { admin =>
+            // topic is not deleted, so it should remain.
+            admin.listTopics().names().get.asScala must (contain(topic))
+
+            var topicInfo: TopicDescription = admin.describeTopics(Collections.singleton(topic)).allTopicNames().get().get(topic)
+            val pl: Seq[TopicPartition] = topicInfo.partitions().asScala.map(info => new TopicPartition(topic, info.partition())).toList
+            val e: Map[TopicPartition, ListOffsetsResult.ListOffsetsResultInfo] = admin.listOffsets(pl.map(tp => tp -> OffsetSpec.earliest()).toMap.asJava).all().get().asScala.toMap
+            val l: Map[TopicPartition, ListOffsetsResult.ListOffsetsResultInfo] = admin.listOffsets(pl.map(tp => tp -> OffsetSpec.latest()).toMap.asJava).all().get().asScala.toMap
+
+            // the earliest offset must not match the latest offset, since all others were deleted, when the topic was truncated.
+            e(new TopicPartition(topic, 0)).offset() mustEqual 2
+            l(new TopicPartition(topic, 0)).offset() mustEqual 2
```

Comment:
```suggestion
            e(new TopicPartition(topic, 0)).offset() must beGreaterThan(1)
            l(new TopicPartition(topic, 0)).offset() must beGreaterThan(1)
```
kafka sometimes skips offsets, so this check isn't reliable. in fact, it might be better to consume from the topic (with auto-offset-reset=earliest) to verify that there aren't any messages, however i guess we'd have to just wait some period of time since it's hard to prove a negative.